projects/qemu.org/package.yml

```diff
@@ -57,17 +57,53 @@ dependencies:
   facebook.com/zstd: '*'
   invisible-island.net/ncurses: '*'
   libssh.org: '*'
+  libpng.org: '*'
 
 build:
   dependencies:
+    #
+    # ../target/hexagon/meson.build:186:8: ERROR: Program 'bison' not found or not executable
+    #
+    gnu.org/bison: '*'
+    #
+    # ../target/hexagon/meson.build:180:8: ERROR: Program 'flex' not found or not executable
+    #
+    github.com/westes/flex: '*'
+    #
+    # python.org: '*'
+    #
+    # build in pkgxdev/pkgx container:
+    #
+    # [279/9472] Generating ui/dbus-display gdbus-codegen with a custom command
+    # FAILED: ui/dbus-display1.h ui/dbus-display1.c 
+    # /root/.pkgx/gnome.org/glib/v2.78.0/lib/pkgconfig/../../bin/gdbus-codegen ui/dbus-display1.xml --glib-min-required 2.64 --output-directory /pkgxdev-pantry/builds/qemu.org-8.1.2+linux/build/ui --interface-prefix org.qemu. --c-namespace QemuDBus --generate-c-code dbus-display1
+    # Traceback (most recent call last):
+    #  File "/root/.pkgx/gnome.org/glib/v2.78.0/lib/pkgconfig/../../bin/gdbus-codegen", line 53, in <module>
+    #    from codegen import codegen_main
+    #  File "/root/.pkgx/gnome.org/glib/v2.78.0/share/glib-2.0/codegen/codegen_main.py", line 29, in <module>
+    #    from . import dbustypes
+    #  File "/root/.pkgx/gnome.org/glib/v2.78.0/share/glib-2.0/codegen/dbustypes.py", line 22, in <module>
+    #    from . import utils
+    #  File "/root/.pkgx/gnome.org/glib/v2.78.0/share/glib-2.0/codegen/utils.py", line 22, in <module>
+    #    import distutils.version
+    # ModuleNotFoundError: No module named 'distutils'
+    #
+    python.org: '<3.10'
     ninja-build.org: '*'
     freedesktop.org/pkg-config: '*'
+
   script:
+    - if: linux
+      run: |
+        sudo apt -o DPkg::Lock::Timeout=60 update
+        sudo apt -o DPkg::Lock::Timeout=60 -y install build-essential
```

Comment:
we definitely _shouldn't_ do this. this is likely to link in OS libraries, which might not be available on out systems. what was this intended to solve?

Reply:
i know but with all the tests i did (version max of python, need of bison/flex), compiling in the docker pkgxdev/pkgx container needed that to compile. better using things from pkgx but i think we are mising some dependencies.

Reply:
right, but this permanently alters the runners, and will have extremely far-reaching impacts on numerous packages. i need to go clean this up.

Reply:
here's the list of things it added:
```
binutils binutils-aarch64-linux-gnu binutils-common cpp cpp-8 dpkg-dev g++ g++-8 gcc gcc-8 libasan5 libatomic1 libbinutils libcc1-0 libdpkg-perl libgcc-8-dev
  libgdbm-compat4 libgdbm6 libgomp1 libisl19 libitm1 liblsan0 libmpc3 libmpfr6 libperl5.28 libstdc++-8-dev libtsan0 libubsan1 make patch perl perl-modules-5.28
```

Reply:
> right, but this permanently alters the runners, and will have extremely far-reaching impacts on numerous packages. i need to go clean this up.

sorry, i thought the builds were isolated.

Reply:
We clean up all the directories used by the builds, but self-hosted runners aren't destroyed and rebuilt every time, sadly. A short coming of needing aarch64 runners.

Reply:
> here's the list of things it added:
> 
> ```
> binutils binutils-aarch64-linux-gnu binutils-common cpp cpp-8 dpkg-dev g++ g++-8 gcc gcc-8 libasan5 libatomic1 libbinutils libcc1-0 libdpkg-perl libgcc-8-dev
>   libgdbm-compat4 libgdbm6 libgomp1 libisl19 libitm1 liblsan0 libmpc3 libmpfr6 libperl5.28 libstdc++-8-dev libtsan0 libubsan1 make patch perl perl-modules-5.28
> ```

well, it seems finally that build-essential is not needed.
it now builds without it in the pkgxdev/pkgx container.